scripts/bash/format.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+for f in $(git grep --cached -Il res -- :/*.hpp :/*.cpp); do
+    clang-format -i $(git rev-parse --show-toplevel)/${f}
```

Comment:
Should we force a specific `clang-format` here, e.g. `clang-format-10`?

Reply:
I thought about that. I don't really want to... because it's possible clang-format will be stable on some version and it's annoying to force the user to pin to a specific version. But if we start seeing incompatible formats on the CI and on people's commits maybe we need to revisit that. What do you think?

Reply:
I agree it's annoying but historically I think `clang-format` has not been totally stable across versions. Maybe our code is un-fancy enough that this isn't an issue though. What about comparing the major version number from `clang-format --version` with some preferred version, like
```bash
CLANG_FORMAT_PREFERRED_VERSION=10
# Regex magic
echo "Warning! clang-format version does not match preferred major version $CLANG_FORMAT_PREFERRED_VERSION!"
```

Reply:
Sure I'll do that.

Reply:
ok looks like my VM is also on 10 so let's pin to that I guess.